src/ripple/app/ledger/InboundLedgers.cpp

```diff
@@ -67,26 +73,29 @@ class InboundLedgersImp
                 auto it = mLedgers.find (hash);
                 if (it != mLedgers.end ())
                 {
-                    // Don't touch failed acquires so they can expire
-                    if (! it->second->isFailed ())
+                    inbound = it->second;
+
+                    if (! inbound->isFailed ())
                     {
-                        it->second->update (seq);
-                        if (it->second->isComplete ())
-                            ret = it->second->getLedger ();
+                        // Don't touch failed acquires so they can expire
```

Comment:
We have a table of ledgers that are "inbound" (i.e. that we're trying to get). We periodically expire old items out of this table. If the acquisition has failed, then we shouldn't update the entry, because it will never expire (it will be seen as recently used) and thus will linger in the table.

I _personally_ think that the existing comment is sufficient, but I'm open to adding a bit more.